docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,85 @@
+---
+lang: en
+title: 'Differences in LoopBack 3 and LoopBack 4 request/response cycle'
+keywords: LoopBack 4.0, LoopBack 4, LoopBack 3.0, LoopBack 3
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/LB3-vs-LB4-request-response-cycle.html
+---
+
+## Differences between LoopBack 3 and LoopBack 4 request/response cycle
+
+The request/response cycle infrastructure and pathway are very different in
+LoopBack 3 and LoopBack 4. Knowing the differences will help you migrate
+LoopBack 3 apps to LoopBack 4 and implement new request/response related
+features in LoopBack 4.
+
+This document will guide you through the differences and show the LoopBack 4
+equivalent, if there is any.
+
+### Request/response infrastructure
+
+The difference begins with the LoopBack object itself. In LoopBack 3, it is an
+instance of an Express app; in LoopBack 4, it is not. Although LoopBack 4 uses
+Express as the HTTP server, it is not directly exposed anymore.
+
+In LoopBack 3 you could add routes and load custom middleware using `app.get()`,
+`app.post()`, `app.use()`, etc., just like how you do in Express.
+In LoopBack 4, you cannot do it anymore. The closest to doing this is to use the
```

Comment:
> In LoopBack 4, you cannot do it anymore

Maybe we can make this more positive. :).  Some suggestion:
In LoopBack 4, a custom Express router can be mounted on a LoopBack app using the RestApplication.mountExpressRouter() API.

Reply:
Facts should be presented as clearly and straightforwardly as possibly, so that expectations are set right there are no misconceptions. If we don't tell a certain popular way of doing something is not supported anymore, we are adding ambiguity. Let's not do that.

Reply:
I agree with Diana that "you cannot do it anymore" leaves negative impression. We have plans to support Express middleware, see https://github.com/strongloop/loopback-next/issues/1293 and https://github.com/strongloop/loopback-next/issues/2035. It is also possible to mount a LB4 app in a new top-level Express app, that's our currently recommended way for installing Express middleware - see https://loopback.io/doc/en/lb4/express-with-lb4-rest-tutorial.html

I am proposing to rephrase the sentence as "you cannot do it yet" and provide users with details about the current status (workaround) and future plans.

Reply:
> I am proposing to rephrase the sentence as "you cannot do it yet"

Even better: _"LoopBack 4 does not provide first-class support for mounting application-wide Express middleware yet."_ (and so on)